### PR TITLE
added about to start and submit handler for generate coversheet event

### PR DESCRIPTION
**Pull request by jhart-dev**
added about to start and submit handler for generate coversheet event

Jira Ticket: [https://tools.hmcts.net/jira/browse/SSCS-8966](url)

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/generatecoversheet/GenerateCoversheetAboutToStartHandler.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            ByteArrayMultipartFile file = ByteArrayMultipartFile.builder()
+                    .content(urlByte.get())
+                    .name(FILENAME)
+                    .contentType(APPLICATION_PDF).build();
+
+            uploadResponse = evidenceManagementService.upload(singletonList(file), DM_STORE_USER_ID);
+        }
+
+        if (nonNull(uploadResponse)) {
+            String location = uploadResponse.getEmbedded().getDocuments().get(0).links.self.href;
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/generatecoversheet/GenerateCoversheetAboutToStartHandler.java:68]: Could throw null pointer if no embedded documents

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/generatecoversheet/GenerateCoversheetAboutToStartHandler.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        UploadResponse uploadResponse = null;
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
+
+        if (urlByte.isPresent()) {
+            ByteArrayMultipartFile file = ByteArrayMultipartFile.builder()
+                    .content(urlByte.get())
+                    .name(FILENAME)
+                    .contentType(APPLICATION_PDF).build();
+
+            uploadResponse = evidenceManagementService.upload(singletonList(file), DM_STORE_USER_ID);
```
Comment (srochani) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/generatecoversheet/GenerateCoversheetAboutToStartHandler.java:64]: Are we handing any error if upload failed? or throws error

Comment (jhart-dev) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/generatecoversheet/GenerateCoversheetAboutToStartHandler.java:64]: Yes, in upload method looks like a HttpClientErrorException is being caught if documentUploadClientApi.upload method fails

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/generatecoversheet/GenerateCoversheetAboutToStartHandler.java (hunk 1 above)
Comment (jhart-dev) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/generatecoversheet/GenerateCoversheetAboutToStartHandler.java:68]: added null and empty checks

**Review by srochani — APPROVED** [review]

**Pull request merged**